### PR TITLE
scaffold e-messaging pages behind feature flag

### PR DESCRIPTION
**Pull request by gabrielsaruhashi**
## Proposed changes
Feature flag fiesta
[sc-1290]

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -96,14 +96,21 @@
...
     "gh-pages": "^3.1.0",
     "husky": "^4.2.5",
     "lint-staged": "^10.2.11",
     "postcss": "^7.0.35",
     "prettier": "^2.0.5",
     "tailwindcss": "npm:@tailwindcss/postcss7-compat@^2.1.2",
-    "typescript": "^3.9.7"
+    "typescript": "^4.1.6"
+  },
+  "resolutions": {
```
Comment (gabrielsaruhashi) [inline package.json:111]: had to update it to take advantage of template literals

File: src/components/Menu/Menu.tsx [hunk 2]
```diff
@@ -49,7 +50,8 @@ export default function Menu({
   callRequestsCount,
 }: Props): ReactElement {
   const [collapsed, setCollapsed] = useState<boolean>(false);
-  const history = useHistory();
+  const navigate = useAppNavigate();
```
Comment (gabrielsaruhashi) [inline src/components/Menu/Menu.tsx:53]: preferred way as it keeps track of history location on the redux store
useHistory was legacy code

File: src/utils/featureEnabled.ts [hunk 3]
```diff
@@ -1,3 +1,6 @@
-export default function featureEnabled(name: string): boolean {
+const features = ['GRAPHQL', 'EMESSAGING'] as const;
```
Comment (gabrielsaruhashi) [inline src/utils/featureEnabled.ts:1]: could not resist the temptation

File: src/pages/MessageReviewPage.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import React from "react";
+import Header from "src/components/Header";
+import featureEnabled from "src/utils/featureEnabled";
+
+interface Props {}
+
+const MessageReviewPage = (props: Props) => {
+  if (!featureEnabled("EMESSAGING")) return <div />;
```
Comment (cainlevy) [inline src/pages/MessageReviewPage.tsx:8]: it's generally fine to only guard the routes

File: src/utils/featureEnabled.ts (hunk 3 above)
Comment (cainlevy) [reply src/utils/featureEnabled.ts:1]: oh hey this is pretty good!

**Review by cainlevy — APPROVED** [review]

**Pull request merged**